### PR TITLE
refactor + test: add OasstApiClient in website with tests

### LINKED ISSUE #204: Contract tests between API and website
As per discussion in #180, the communication between the website and API should be tested through a Contract test. This test should test that each call satisfies the OpenAPI schema inputs and endpoints. That way, changes to the endpoint contract will automatically be caught by these tests.

### PR DESCRIPTION
**Pull request by jack-michaud**
Fixes #204 

- refactor: move new task's oasst api fetching into OasstApiClient
- test: add contract tests for fetchTask and ackTask
- docs: Add TODOs to refactor and add more tests
  - #353 and #354
- ci: run frontend contract tests in Test API Contract workflow

### CODE REVIEW

File: website/src/pages/api/update_task.ts
Comment (fozziethebeat) [inline website/src/pages/api/update_task.ts:40]: Are you tackling this in a follow up PR?

**Review by fozziethebeat — APPROVED** [review]
This looks good.  I'm approving given that you'll need others to review before it's ready to merge.  Please update the comment to reference the issue and ensure the issue isn't closed yet.

File: website/src/lib/oasst_api_client.ts
Comment (AbdBarho) [inline website/src/lib/oasst_api_client.ts:34]: quick question, why cloning here?

File: website/src/pages/api/update_task.ts
Comment (jack-michaud) [reply website/src/pages/api/update_task.ts:40]: bdcfae5

File: website/src/lib/oasst_api_client.ts
Comment (AbdBarho) [reply website/src/lib/oasst_api_client.ts:34]: Aha! makes sense.

I guess using the `content-type` of the response is not reliable enough?

would it make sense to do it like this
```js
const text = await resp.text();
try {
  const error = JSON.parse(text);
  throw new OasstError(error.message, error.error_code, resp.status)
} catch (e) {
  // not json
  throw new OasstError(text, 0, resp.status);
}
```
this is equivalent right?

Comment (jack-michaud) [reply website/src/lib/oasst_api_client.ts:34]: Truth be told, I didn't think to use `content-type`. Using a `try` `catch` is a little more defensive. 

The way you've suggested is clearer and equivalent. I'll switch it!

Comment (jack-michaud) [reply website/src/lib/oasst_api_client.ts]: 695e1f3

**Review by AbdBarho — APPROVED** [review]
Thanks! waiting on non-web devs approval.

**Review by yk — APPROVED** [review]
thank you very much, amazing!

**Pull request merged**
